### PR TITLE
migrate java8 to java11

### PR DESCRIPTION
**Pull request by kaibocai**
### Issue describing the changes in this PR

migrate java8 to java11 worker logics. Remove redundant codes. 

### Pull request checklist

* [X] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [X] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [X] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/test/java/com/microsoft/azure/functions/worker/broker/JavaMethodExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+public class JavaMethodExecutorTest {
+
+	private static FunctionClassLoaderProvider functionClassLoaderProvider = new FunctionClassLoaderProvider();
+
+	//TODO: figure out why below logics not always works on azure devops pipelines, it seems more to be a issue from
+	//	azure devops pipelines. Currently, all test cases are testing the TestFunctionsClass using system classloader as
+	// TestFunctionsClass is in the classpath when running the unit tests.
+
+//	@BeforeClass
```
Comment (shreyas-gopalakrishna) [inline src/test/java/com/microsoft/azure/functions/worker/broker/JavaMethodExecutorTest.java]: Let us discuss about this test case and format this class.

**Review by shreyas-gopalakrishna — DISMISSED** [review]
LGTM!

**Comment by kaibocai** [comment]
Hi Team, @shreyas-gopalakrishna , @amamounelsayed , @trask , this PR is ready for review now, please take a look, thanks very much!

File: src/test/java/com/microsoft/azure/functions/worker/broker/JavaMethodExecutorTest.java (hunk 1 above)
Comment (kaibocai) [reply src/test/java/com/microsoft/azure/functions/worker/broker/JavaMethodExecutorTest.java]: This test case issue was resolved by adding logics to reset the singleton object before test case.

**Review by shreyas-gopalakrishna — APPROVED** [review]

**Pull request merged**